internal/app.go

```diff
@@ -66,3 +72,24 @@ func (a *App) Run() error {
 
 	return nil
 }
+
+// handleSignals handles signals to attempt graceful shutdown
+func handleSignals(
```

Comment:
Why not just manage the lifecycle of `App` with the parent context? We could just use `signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)` on line 42 and the context would automatically get cancelled when we receive an interrupt signal.

We could then pass that context around to child goroutines, who would then get cancelled if the parent context is cancelled.